### PR TITLE
Add SimpleQueue and a unit test.

### PR DESCRIPTION
**Pull request by donndenman**
Merry Christmas Bill!

Feel free to critique or send code review comments!

### CODE REVIEW

File: collections.js [hunk 1]
```diff
@@ -22,25 +22,25 @@ function LinkedList() {
 
 LinkedList.prototype.empty = function() {
   return this.length == 0;
-}
+};
```
Comment (billbudge) [inline collections.js:25]: Why add semicolons here?

File: collections.js [hunk 2]
```diff
@@ -118,8 +118,36 @@ LinkedList.prototype.find = function(value) {
     node = node.next;
   }
   return null;
+};
+
+//------------------------------------------------------------------------------
+// Simple Queue.
+
+function SimpleQueue() {
```
Comment (billbudge) [inline collections.js:126]: I would just call this Queue. It's more consistent with other languages (C++, Java, C#, etc.)

File: collections.js [hunk 3]
```diff
@@ -118,8 +118,36 @@ LinkedList.prototype.find = function(value) {
...
 }
 
+SimpleQueue.prototype = {
+  enqueue: function(item) {
+    this.q_.push(item);
+    return this;
+  },
+
+  dequeue: function() {
+    return this.q_.shift();
```
Comment (billbudge) [inline collections.js:137]: I initially thought that Array.shift() would be too slow, but apparently it's not that bad:
http://stackoverflow.com/questions/8423493/what-is-the-performance-of-objects-arrays-in-javascript-specifically-for-googl

File: collections.js [hunk 4]
```diff
@@ -118,8 +118,36 @@ LinkedList.prototype.find = function(value) {
...
+  },
+
+  empty: function() {
+    return this.q_.length === 0;
+  },
+  
+  flush: function() {
```
Comment (billbudge) [inline collections.js:144]: clear() for consistency with other collections here and JS builtin collections (Map, Set)

File: collections.js [hunk 5]
```diff
@@ -118,8 +118,36 @@ LinkedList.prototype.find = function(value) {
...
+    var result = this.q_.slice(0);
+	this.q_.length = 0;
+	return result;
```
Comment (billbudge) [inline collections.js:147]: Why not just reassign the array? Also, it's strange to return the array, which is an implementation detail.

  this.q_ = [];
  return this;

File: collections_tests.js [hunk 6]
```diff
@@ -130,6 +130,49 @@ test("LinkedList find", function() {
 });
 
 //------------------------------------------------------------------------------
+// SimpleQueue tests.
+
+test("SimpleQueue basic operation", function() {
+  var q = new diagrammar.collections.SimpleQueue();
```
Comment (billbudge) [inline collections_tests.js:136]: Test empty, dequeue here.

File: collections.js (hunk 1 above)
Comment (donndenman) [inline collections.js:25]: These are not function declarations, rather they are regular assignment statements (that happen to use function literals) and all assignment statements should end in a semicolon.

File: collections.js (hunk 2 above)
Comment (donndenman) [inline collections.js:126]: Darn, this isn't my latest code.  In my latest, I just use Queue.  I'll update to send you my latest... hopefully soon!

File: collections.js (hunk 3 above)
Comment (donndenman) [inline collections.js:137]: Thanks for the info -- good to know!

**Comment by donndenman** [comment]
Here are a few comments, but I'll send you a new request ... hopefully quite soon.

**Pull request closed without merging**